uaclient/clouds/tests/test_aws.py

```diff
@@ -17,6 +20,44 @@ def test_identity_doc_from_aws_url_pkcs7(self, readurl):
         url = "http://169.254.169.254/latest/dynamic/instance-identity/pkcs7"
         assert [mock.call(url)] == readurl.call_args_list
 
+    @pytest.mark.parametrize("caplog_text", [logging.DEBUG], indirect=True)
+    @pytest.mark.parametrize(
+        "fail_count,exception", ((3, None), (4, HTTPError))
+    )
+    @mock.patch(M_PATH + "util.time.sleep")
+    @mock.patch(M_PATH + "util.readurl")
+    def test_retry_backoff_on_failed_identity_doc(
+        self, readurl, sleep, fail_count, exception, caplog_text
+    ):
+        """Retry backoff is attempted before failing to get AWS.identity_doc"""
+
+        def fake_someurlerrors(url):
+            if readurl.call_count <= fail_count:
+                raise HTTPError(
+                    None, 400 + readurl.call_count, None, None, BytesIO()
```

Comment:
A nit, but I would suggest using something other than 400 as the starting point here. When I see "404" and "HTTP Error 401" those have semantic meaning that isn't relevant here. Whereas, say, 704 and "HTTP Error 701" are more obviously just test data.

Reply:
very good point. fixed.